lib/cloud_controller/runner.rb

```diff
@@ -167,10 +167,10 @@ def periodic_updater
         Time.now.utc,
         @log_counter,
         Steno.logger('cc.api'),
-        [
-          VCAP::CloudController::Metrics::StatsdUpdater.new(statsd_client)
-        ]
+        VCAP::CloudController::Metrics::StatsdUpdater.new(statsd_client),
+        prometheus_updater
       )
+      CloudController::DependencyLocator.instance.register(:periodic_updater, @periodic_updater)
```

Comment:
I did that because the `statsd_client`, passed to the constructor of the `PeriodicUpdater` has another configuration than the one in `DependencyLocator`. And also other instances like `log_counter` are passed. Using parameters for  `DependencyLocator.instance.periodic_updater` to configure the `PeriodicUpdater` feels not right to me. So I thought it's okay to initialise it here and then register it with the `DependencyLocator` for the rest of the code. I think it would make sense to look at all those metric/log clients, which seem to get initialised all over the code and check whether it would make sense to store a singleton in the `DependencyLocator` for all of those. But that's probably then another PR.

Reply:
Good point, I adjusted it accordingly.